datastore/importers/nmdb-import-systeminfo/Parser.cpp

```diff
@@ -0,0 +1,167 @@
+// =============================================================================
+// Copyright 2024 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+#include "Parser.hpp"
+
+// =============================================================================
+// Parser logic
+// =============================================================================
+Parser::Parser() : Parser::base_type(start)
+{
+    start =
+        (*qi::eol > systeminfo > -qi::eol)[(qi::_val = pnx::bind(&Parser::getData, this))];
+
+    token =
+        +qi::ascii::graph;
+
+    host_name =
+        ("Host Name: " > token[pnx::bind(&nmdo::DeviceInformation::setDeviceId, &data.devInfo, qi::_1)] > qi::eol);
```

Comment:
Ahh okay, I figured I would take the approach of just parsing until we the match and then grab the information then and continue the rest of the rule, If its okay with you, I have the parser working and I would end up spending more time trying to troubleshoot changing all these rules

Reply:
Sure.  As I said, you don't have to make the change.

Just of note, using semantic actions can cause problems and, your described usage case, ultimately may make you run into more problems than necessary.  This is primarily because of the state change they cause.  Luckily, you aren't allowing backtracking so there isn't much of a concern as if the rule logic fails, the whole parser fails.